modules/javafx.base/src/test/java/test/javafx/beans/property/ObjectPropertyLeakTest.java

```diff
@@ -0,0 +1,278 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.javafx.beans.property;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.DoubleProperty;
+import javafx.beans.property.FloatProperty;
+import javafx.beans.property.IntegerProperty;
+import javafx.beans.property.LongProperty;
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.Property;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.beans.property.SimpleDoubleProperty;
+import javafx.beans.property.SimpleFloatProperty;
+import javafx.beans.property.SimpleIntegerProperty;
+import javafx.beans.property.SimpleLongProperty;
+import javafx.beans.property.SimpleObjectProperty;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class ObjectPropertyLeakTest {
+
+    private static final int OBJ_COUNT = 10;
+
+    private final ArrayList<Property<?>> origList = new ArrayList<>();
+    private final ArrayList<Property<?>> wrappedList = new ArrayList<>();
+
+    private final ArrayList<WeakReference<Property<?>>> origRefs = new ArrayList<>();
+    private final ArrayList<WeakReference<Property<?>>> wrappedRefs = new ArrayList<>();
+
+    private void commonLeakTest(int origExpected, int wrappedExpected)
+            throws Exception {
+
+        for (int i = 0; i < 5; i++) {
+            System.gc();
+            System.runFinalization();
+            Thread.sleep(50);
+        }
+
+        int origCount = 0;
+        for (var ref : origRefs) {
+            if (ref.get() != null) origCount++;
+        }
+        final String origMsg = origExpected > 0
+                ? "Original properties should NOT be GCed"
+                : "Original properties should be GCed";
+        assertEquals(origMsg, origExpected, origCount);
```

Comment:
As long as I'm, modifying the test, I'll also move the above block, which is repeated twice, into a separate method.